diff --git a/TSS_LMS_BE/src/helpers/getClassesByUserId.js b/TSS_LMS_BE/src/helpers/getClassesByUserId.js
index 85c352a..194dbc6 100644
--- a/TSS_LMS_BE/src/helpers/getClassesByUserId.js
+++ b/TSS_LMS_BE/src/helpers/getClassesByUserId.js
@@ -8,10 +8,6 @@ export const getUserData = async (userId) => {
                             AND usuario.id=?`;
   const [rows] = await dbConnection.query(query, [userId]);
 
-  if (rows.length === 0) {
-    return null;
-  }
-//toDo
   const classes = rows.map((item) => {
     return {
       id_clase: item.id_clase,
diff --git a/TSS_LMS_BE/src/index.js b/TSS_LMS_BE/src/index.js
index f7bab6b..741e9e2 100644
--- a/TSS_LMS_BE/src/index.js
+++ b/TSS_LMS_BE/src/index.js
@@ -8,7 +8,7 @@ import cors from 'cors'
 import practiceRoutes from './routes/PracticeRoutes.js';
 import authRoutes from './routes/auth.js';
 import contriRoutes from './routes/ContriRoutes.js';
-import forumRoutes from './routes/forumRoutes.js';
+import forumRoutes from './routes/ForumRoutes.js';
 import contributionMessageRoutes from './routes/ContributionMessageRoutes.js';
 import messageRoutes from './routes/messagesRoutes.js';
 import taskRoutes from './routes/taskRoutes.js';
diff --git a/TSS_LMS_BE/src/routes/ChapterRoutes.js b/TSS_LMS_BE/src/routes/ChapterRoutes.js
index 76135a6..e63b610 100644
--- a/TSS_LMS_BE/src/routes/ChapterRoutes.js
+++ b/TSS_LMS_BE/src/routes/ChapterRoutes.js
@@ -41,6 +41,10 @@ chapterRoutes.get("/chapters/:id/material", async (req, res) => {
 });
 
 chapterRoutes.get("/chapters/:id/forum", async (req, res) => {
+  const [chapters] = await dbConnection.query(
+    `SELECT * FROM capitulo WHERE id=?`,
+    [req.params.id]
+  );
   const [rows, fields] = await dbConnection.query(
     `
     SELECT  capitulo.titulo_capitulo as titulo_capitulo,
@@ -64,17 +68,21 @@ chapterRoutes.get("/chapters/:id/forum", async (req, res) => {
     };
   });
   const result = {
-    titulo_capitulo: rows[0].titulo_capitulo,
-    titulo_material: rows[0].titulo_material,
-    estado: rows[0].estado,
-    titulo_foro: rows[0].titulo_foro,
-    descripcion_foro: rows[0].descripcion_foro,
+    titulo_capitulo: chapters[0].titulo_capitulo,
+    titulo_material: chapters[0].titulo_material,
+    estado: chapters[0].estado,
+    titulo_foro: chapters[0].titulo_foro,
+    descripcion_foro: chapters[0].descripcion_foro,
     mensajes: messages,
   };
   res.send(result);
 });
 
 chapterRoutes.get("/chapters/:id/practice", async (req, res) => {
+  const [chapters] = await dbConnection.query(
+    `SELECT * FROM capitulo WHERE id=?`,
+    [req.params.id]
+  );
   const [rows, fields] = await dbConnection.query(
     `SELECT 	capitulo.id as id_capitulo ,
                                                             capitulo.titulo_capitulo as titulo_capitulo,
@@ -97,8 +105,8 @@ chapterRoutes.get("/chapters/:id/practice", async (req, res) => {
   });
 
   const result = {
-    id_capitulo: rows[0].id_capitulo,
-    titulo_capitulo: rows[0].titulo_capitulo,
+    id_capitulo: chapters[0].id_capitulo,
+    titulo_capitulo: chapters[0].titulo_capitulo,
     practicas: practices,
   };
 
diff --git a/TSS_LMS_BE/src/routes/CompilerRoutes.js b/TSS_LMS_BE/src/routes/CompilerRoutes.js
index 8fced54..d6ed2dc 100644
--- a/TSS_LMS_BE/src/routes/CompilerRoutes.js
+++ b/TSS_LMS_BE/src/routes/CompilerRoutes.js
@@ -2,6 +2,7 @@ import {Router} from 'express';
 import fs from 'fs';
 import {exec} from 'child_process';
 import { v4 as uuidv4 } from 'uuid';
+import path from 'path';
 
 const CompilerRoutes= Router();
 //toDo refactorizar
@@ -15,8 +16,12 @@ CompilerRoutes.post('/compiler',async(req,res)=>{
     fs.mkdirSync(excecutionDirectory,{recursive:true});
 
     fs.writeFile(javaFileDir, code, (err) => {if (err) throw err;}); 
-
-    exec(`javac ${javaFileDir} && java -cp ${excecutionDirectory} ${className}`, (err, stdout) => {if (err) {return;}
+    const absPath = path.resolve(javaFileDir);
+    const directory = path.dirname(absPath);
+    exec(`javac ${absPath} && java -cp ${directory} ${className}`, (err, stdout) => {if (err) {
+        console.log(err.message)
+        return;
+    }
     res.send(stdout)
     });
 
diff --git a/TSS_LMS_BE/src/routes/taskRoutes.js b/TSS_LMS_BE/src/routes/taskRoutes.js
index e1c372b..e614ace 100644
--- a/TSS_LMS_BE/src/routes/taskRoutes.js
+++ b/TSS_LMS_BE/src/routes/taskRoutes.js
@@ -97,7 +97,7 @@ taskRoutes.put("/tasks/responses",async(req,res)=>{
     const {id_usuario,id_tarea,nota}=req.body;
 
     const updateNoteQuery=`UPDATE respuesta_tarea SET nota = ? 
-                               WHERE id_usuario = 2 AND id_tarea = ?`;
+                               WHERE id_usuario = ? AND id_tarea = ?`;
 
     await dbConnection.query(updateNoteQuery,[nota,id_usuario,id_tarea]);
 
diff --git a/TSS_LMS_FE/src/EducativePlatform/components/ChapterLayout/ChapterLayout.jsx b/TSS_LMS_FE/src/EducativePlatform/components/ChapterLayout/ChapterLayout.jsx
index c454b84..78c586b 100644
--- a/TSS_LMS_FE/src/EducativePlatform/components/ChapterLayout/ChapterLayout.jsx
+++ b/TSS_LMS_FE/src/EducativePlatform/components/ChapterLayout/ChapterLayout.jsx
@@ -98,7 +98,7 @@ const editarCapitulo=(e)=>{
             <button
               className="text-light py-2 px-3  d-block my-3"
               style={{ backgroundColor: "#1D2939", borderRadius: "20px" }}
-              onClick={()=>{navigate(`foro`)}}>
+              onClick={()=>{navigate(`../${id_capitulo}/foro`)}}>
               Revisar
             </button>
           </div>
diff --git a/TSS_LMS_FE/src/EducativePlatform/components/RunCodeBtn/RunCodeBtn.jsx b/TSS_LMS_FE/src/EducativePlatform/components/RunCodeBtn/RunCodeBtn.jsx
index f80c8ea..376e25e 100644
--- a/TSS_LMS_FE/src/EducativePlatform/components/RunCodeBtn/RunCodeBtn.jsx
+++ b/TSS_LMS_FE/src/EducativePlatform/components/RunCodeBtn/RunCodeBtn.jsx
@@ -14,8 +14,7 @@ export const RunCodeBtn = ({code,className,onOutputChange}) => {
 
 
       const response=await petition.json();
-      console.log(response)
-      onOutputChange(response.res)
+      onOutputChange(response)
     }
     
       return (
