diff --git a/TSS_LMS_BE/.gitignore b/TSS_LMS_BE/.gitignore
index 07e6e47..371fce9 100644
--- a/TSS_LMS_BE/.gitignore
+++ b/TSS_LMS_BE/.gitignore
@@ -1 +1,2 @@
 /node_modules
+/assets
diff --git a/TSS_LMS_BE/assets/06406660-5d36-4fdd-85d3-715a11302256/Ejercicio2.class b/TSS_LMS_BE/assets/06406660-5d36-4fdd-85d3-715a11302256/Ejercicio2.class
deleted file mode 100644
index 688cdb2..0000000
Binary files a/TSS_LMS_BE/assets/06406660-5d36-4fdd-85d3-715a11302256/Ejercicio2.class and /dev/null differ
diff --git a/TSS_LMS_BE/assets/06406660-5d36-4fdd-85d3-715a11302256/Ejercicio2.java b/TSS_LMS_BE/assets/06406660-5d36-4fdd-85d3-715a11302256/Ejercicio2.java
deleted file mode 100644
index b9d563e..0000000
--- a/TSS_LMS_BE/assets/06406660-5d36-4fdd-85d3-715a11302256/Ejercicio2.java
+++ /dev/null
@@ -1,43 +0,0 @@
-class Ejercicio2 {
-    public static void main(String[] args) {
-        System.out.println(generarValorSimulado(1, 2, 3));
-    }
-
-    public static double generarValorSimulado(double pesimista, double probable, double optimista) {
-        double a = pesimista;
-        double b = probable;
-        double c = optimista;
-        double x = 0;
-        
-        congruencialMixto generador = new congruencialMixto();
-        double R = generador.generarValorAleatorio();
-        
-        if (R <= ((b - a) / (c - a))) {
-            x = c - Math.sqrt((c - a) * (b - a) * R);
-        } else {
-            x = c - Math.sqrt((c - a) * (c - b) * (1 - R));
-        }
-        
-        return x;
-    }
-}
-
-class congruencialMixto {
-    public static double generarValor(int semilla, int cmultiplicativa, int modulo) {
-        int i, numero;
-        double numero2 = 0;
-        
-        for (i = 0; i <= 20; i++) {
-            numero = (cmultiplicativa * semilla) % modulo;
-            numero2 = (double) numero / (double) (modulo - 1);
-            semilla = numero;
-        }
-        
-        return numero2;
-    }
-    
-    public static double generarValorAleatorio() {
-        double numero = (double) (Math.random());
-        return numero;
-    }
-}
\ No newline at end of file
diff --git a/TSS_LMS_BE/assets/06406660-5d36-4fdd-85d3-715a11302256/congruencialMixto.class b/TSS_LMS_BE/assets/06406660-5d36-4fdd-85d3-715a11302256/congruencialMixto.class
deleted file mode 100644
index 2545495..0000000
Binary files a/TSS_LMS_BE/assets/06406660-5d36-4fdd-85d3-715a11302256/congruencialMixto.class and /dev/null differ
diff --git a/TSS_LMS_BE/assets/1da5822d-6f4b-48b9-9faf-680ddd086d01/HolaMundo.class b/TSS_LMS_BE/assets/1da5822d-6f4b-48b9-9faf-680ddd086d01/HolaMundo.class
deleted file mode 100644
index 2dc3a9a..0000000
Binary files a/TSS_LMS_BE/assets/1da5822d-6f4b-48b9-9faf-680ddd086d01/HolaMundo.class and /dev/null differ
diff --git a/TSS_LMS_BE/assets/1da5822d-6f4b-48b9-9faf-680ddd086d01/HolaMundo.java b/TSS_LMS_BE/assets/1da5822d-6f4b-48b9-9faf-680ddd086d01/HolaMundo.java
deleted file mode 100644
index 40fd1b9..0000000
--- a/TSS_LMS_BE/assets/1da5822d-6f4b-48b9-9faf-680ddd086d01/HolaMundo.java
+++ /dev/null
@@ -1,5 +0,0 @@
-public class HolaMundo {
-  public static void main(String[] args) {
-    System.out.println("Hola, Mundo");
-  }
-}
\ No newline at end of file
diff --git a/TSS_LMS_BE/assets/4598b65d-55a8-4a6b-8b43-9765ae951b41/Ejemplo de numero aleatorio no -Uniforme.java b/TSS_LMS_BE/assets/4598b65d-55a8-4a6b-8b43-9765ae951b41/Ejemplo de numero aleatorio no -Uniforme.java
deleted file mode 100644
index d1d7bd0..0000000
--- a/TSS_LMS_BE/assets/4598b65d-55a8-4a6b-8b43-9765ae951b41/Ejemplo de numero aleatorio no -Uniforme.java	
+++ /dev/null
@@ -1,7 +0,0 @@
-public class CountToNine {
-    public static void main(String[] args) {
-        for (int i = 1; i <= 9; i++) {
-            System.out.println(i);
-        }
-    }
-}
diff --git a/TSS_LMS_BE/assets/493ced01-9f7d-4f52-a1b4-4fbec0a3a98c/Holamundo.java b/TSS_LMS_BE/assets/493ced01-9f7d-4f52-a1b4-4fbec0a3a98c/Holamundo.java
deleted file mode 100644
index 40fd1b9..0000000
--- a/TSS_LMS_BE/assets/493ced01-9f7d-4f52-a1b4-4fbec0a3a98c/Holamundo.java
+++ /dev/null
@@ -1,5 +0,0 @@
-public class HolaMundo {
-  public static void main(String[] args) {
-    System.out.println("Hola, Mundo");
-  }
-}
\ No newline at end of file
diff --git a/TSS_LMS_BE/assets/72d50062-2e18-49ba-b62e-38f7119a9d10/Ejercicio2.class b/TSS_LMS_BE/assets/72d50062-2e18-49ba-b62e-38f7119a9d10/Ejercicio2.class
deleted file mode 100644
index 688cdb2..0000000
Binary files a/TSS_LMS_BE/assets/72d50062-2e18-49ba-b62e-38f7119a9d10/Ejercicio2.class and /dev/null differ
diff --git a/TSS_LMS_BE/assets/72d50062-2e18-49ba-b62e-38f7119a9d10/Ejercicio2.java b/TSS_LMS_BE/assets/72d50062-2e18-49ba-b62e-38f7119a9d10/Ejercicio2.java
deleted file mode 100644
index b9d563e..0000000
--- a/TSS_LMS_BE/assets/72d50062-2e18-49ba-b62e-38f7119a9d10/Ejercicio2.java
+++ /dev/null
@@ -1,43 +0,0 @@
-class Ejercicio2 {
-    public static void main(String[] args) {
-        System.out.println(generarValorSimulado(1, 2, 3));
-    }
-
-    public static double generarValorSimulado(double pesimista, double probable, double optimista) {
-        double a = pesimista;
-        double b = probable;
-        double c = optimista;
-        double x = 0;
-        
-        congruencialMixto generador = new congruencialMixto();
-        double R = generador.generarValorAleatorio();
-        
-        if (R <= ((b - a) / (c - a))) {
-            x = c - Math.sqrt((c - a) * (b - a) * R);
-        } else {
-            x = c - Math.sqrt((c - a) * (c - b) * (1 - R));
-        }
-        
-        return x;
-    }
-}
-
-class congruencialMixto {
-    public static double generarValor(int semilla, int cmultiplicativa, int modulo) {
-        int i, numero;
-        double numero2 = 0;
-        
-        for (i = 0; i <= 20; i++) {
-            numero = (cmultiplicativa * semilla) % modulo;
-            numero2 = (double) numero / (double) (modulo - 1);
-            semilla = numero;
-        }
-        
-        return numero2;
-    }
-    
-    public static double generarValorAleatorio() {
-        double numero = (double) (Math.random());
-        return numero;
-    }
-}
\ No newline at end of file
diff --git a/TSS_LMS_BE/assets/72d50062-2e18-49ba-b62e-38f7119a9d10/congruencialMixto.class b/TSS_LMS_BE/assets/72d50062-2e18-49ba-b62e-38f7119a9d10/congruencialMixto.class
deleted file mode 100644
index 2545495..0000000
Binary files a/TSS_LMS_BE/assets/72d50062-2e18-49ba-b62e-38f7119a9d10/congruencialMixto.class and /dev/null differ
diff --git a/TSS_LMS_BE/assets/8280e6fe-fb3c-48bc-b56c-82e3a4ee92a3/HolaMundo.class b/TSS_LMS_BE/assets/8280e6fe-fb3c-48bc-b56c-82e3a4ee92a3/HolaMundo.class
deleted file mode 100644
index 2dc3a9a..0000000
Binary files a/TSS_LMS_BE/assets/8280e6fe-fb3c-48bc-b56c-82e3a4ee92a3/HolaMundo.class and /dev/null differ
diff --git a/TSS_LMS_BE/assets/8280e6fe-fb3c-48bc-b56c-82e3a4ee92a3/HolaMundo.java b/TSS_LMS_BE/assets/8280e6fe-fb3c-48bc-b56c-82e3a4ee92a3/HolaMundo.java
deleted file mode 100644
index 40fd1b9..0000000
--- a/TSS_LMS_BE/assets/8280e6fe-fb3c-48bc-b56c-82e3a4ee92a3/HolaMundo.java
+++ /dev/null
@@ -1,5 +0,0 @@
-public class HolaMundo {
-  public static void main(String[] args) {
-    System.out.println("Hola, Mundo");
-  }
-}
\ No newline at end of file
diff --git a/TSS_LMS_BE/assets/ad1dd2c2-3a75-4f64-a91b-1dcb9fb27d1a/SumNumbers.class b/TSS_LMS_BE/assets/ad1dd2c2-3a75-4f64-a91b-1dcb9fb27d1a/SumNumbers.class
deleted file mode 100644
index 11c8c66..0000000
Binary files a/TSS_LMS_BE/assets/ad1dd2c2-3a75-4f64-a91b-1dcb9fb27d1a/SumNumbers.class and /dev/null differ
diff --git a/TSS_LMS_BE/assets/ad1dd2c2-3a75-4f64-a91b-1dcb9fb27d1a/SumNumbers.java b/TSS_LMS_BE/assets/ad1dd2c2-3a75-4f64-a91b-1dcb9fb27d1a/SumNumbers.java
deleted file mode 100644
index 87c631f..0000000
--- a/TSS_LMS_BE/assets/ad1dd2c2-3a75-4f64-a91b-1dcb9fb27d1a/SumNumbers.java
+++ /dev/null
@@ -1,16 +0,0 @@
-class SumNumbers {
-    public static void main(String[] args) {
-        System.out.println(calcularSuma(1, 10));
-    }
-    
-    public static int calcularSuma(int inicio, int fin) {
-        int sum = 0;
-        
-        // Sumar los números desde 'inicio' hasta 'fin'
-        for (int i = inicio; i <= fin; i++) {
-            sum += i;
-        }
-        
-        return sum;
-    }
-}
\ No newline at end of file
diff --git a/TSS_LMS_BE/assets/d1ec8921-dc34-4c0c-9a26-5856bf0526b4/Ejemplo de numero aleatorio no -Uniforme.java b/TSS_LMS_BE/assets/d1ec8921-dc34-4c0c-9a26-5856bf0526b4/Ejemplo de numero aleatorio no -Uniforme.java
deleted file mode 100644
index d1d7bd0..0000000
--- a/TSS_LMS_BE/assets/d1ec8921-dc34-4c0c-9a26-5856bf0526b4/Ejemplo de numero aleatorio no -Uniforme.java	
+++ /dev/null
@@ -1,7 +0,0 @@
-public class CountToNine {
-    public static void main(String[] args) {
-        for (int i = 1; i <= 9; i++) {
-            System.out.println(i);
-        }
-    }
-}
diff --git a/TSS_LMS_BE/assets/d81f6a8d-9c2a-445c-ba18-831960aff39e/HolaMundo.class b/TSS_LMS_BE/assets/d81f6a8d-9c2a-445c-ba18-831960aff39e/HolaMundo.class
deleted file mode 100644
index 2dc3a9a..0000000
Binary files a/TSS_LMS_BE/assets/d81f6a8d-9c2a-445c-ba18-831960aff39e/HolaMundo.class and /dev/null differ
diff --git a/TSS_LMS_BE/assets/d81f6a8d-9c2a-445c-ba18-831960aff39e/HolaMundo.java b/TSS_LMS_BE/assets/d81f6a8d-9c2a-445c-ba18-831960aff39e/HolaMundo.java
deleted file mode 100644
index 40fd1b9..0000000
--- a/TSS_LMS_BE/assets/d81f6a8d-9c2a-445c-ba18-831960aff39e/HolaMundo.java
+++ /dev/null
@@ -1,5 +0,0 @@
-public class HolaMundo {
-  public static void main(String[] args) {
-    System.out.println("Hola, Mundo");
-  }
-}
\ No newline at end of file
diff --git a/TSS_LMS_BE/assets/fc527e17-5b59-42a3-b451-af10a3b10da0/Ejercicio2.class b/TSS_LMS_BE/assets/fc527e17-5b59-42a3-b451-af10a3b10da0/Ejercicio2.class
deleted file mode 100644
index 688cdb2..0000000
Binary files a/TSS_LMS_BE/assets/fc527e17-5b59-42a3-b451-af10a3b10da0/Ejercicio2.class and /dev/null differ
diff --git a/TSS_LMS_BE/assets/fc527e17-5b59-42a3-b451-af10a3b10da0/Ejercicio2.java b/TSS_LMS_BE/assets/fc527e17-5b59-42a3-b451-af10a3b10da0/Ejercicio2.java
deleted file mode 100644
index b9d563e..0000000
--- a/TSS_LMS_BE/assets/fc527e17-5b59-42a3-b451-af10a3b10da0/Ejercicio2.java
+++ /dev/null
@@ -1,43 +0,0 @@
-class Ejercicio2 {
-    public static void main(String[] args) {
-        System.out.println(generarValorSimulado(1, 2, 3));
-    }
-
-    public static double generarValorSimulado(double pesimista, double probable, double optimista) {
-        double a = pesimista;
-        double b = probable;
-        double c = optimista;
-        double x = 0;
-        
-        congruencialMixto generador = new congruencialMixto();
-        double R = generador.generarValorAleatorio();
-        
-        if (R <= ((b - a) / (c - a))) {
-            x = c - Math.sqrt((c - a) * (b - a) * R);
-        } else {
-            x = c - Math.sqrt((c - a) * (c - b) * (1 - R));
-        }
-        
-        return x;
-    }
-}
-
-class congruencialMixto {
-    public static double generarValor(int semilla, int cmultiplicativa, int modulo) {
-        int i, numero;
-        double numero2 = 0;
-        
-        for (i = 0; i <= 20; i++) {
-            numero = (cmultiplicativa * semilla) % modulo;
-            numero2 = (double) numero / (double) (modulo - 1);
-            semilla = numero;
-        }
-        
-        return numero2;
-    }
-    
-    public static double generarValorAleatorio() {
-        double numero = (double) (Math.random());
-        return numero;
-    }
-}
\ No newline at end of file
diff --git a/TSS_LMS_BE/assets/fc527e17-5b59-42a3-b451-af10a3b10da0/congruencialMixto.class b/TSS_LMS_BE/assets/fc527e17-5b59-42a3-b451-af10a3b10da0/congruencialMixto.class
deleted file mode 100644
index 2545495..0000000
Binary files a/TSS_LMS_BE/assets/fc527e17-5b59-42a3-b451-af10a3b10da0/congruencialMixto.class and /dev/null differ
diff --git "a/TSS_LMS_BE/assets/fd071e1d-bd15-4185-8a7e-9c0a1eecd8e4/Ejemplo b\303\241sico de simulaci\303\263n .java" "b/TSS_LMS_BE/assets/fd071e1d-bd15-4185-8a7e-9c0a1eecd8e4/Ejemplo b\303\241sico de simulaci\303\263n .java"
deleted file mode 100644
index baeb734..0000000
--- "a/TSS_LMS_BE/assets/fd071e1d-bd15-4185-8a7e-9c0a1eecd8e4/Ejemplo b\303\241sico de simulaci\303\263n .java"	
+++ /dev/null
@@ -1,5 +0,0 @@
-public class PrintSix {
-    public static void main(String[] args) {
-        System.out.println(6+" Como vemos salio un numero");
-    }
-}
diff --git a/TSS_LMS_BE/src/routes/taskRoutes.js b/TSS_LMS_BE/src/routes/taskRoutes.js
index 0aee7ed..f8f5c82 100644
--- a/TSS_LMS_BE/src/routes/taskRoutes.js
+++ b/TSS_LMS_BE/src/routes/taskRoutes.js
@@ -20,17 +20,18 @@ taskRoutes.get("/chapter/:id/tasks-notes",async(req,res)=>{
 
     const getTasksQuery=`SELECT
                          usuario.id,
-                         usuario.nombre_completo , 
+                         usuario.nombre_completo ,
                          usuario.email,
                          capitulo.titulo_capitulo AS titulo_capitulo,
                          tarea.descripcion AS descripcion_tarea,
-                        respuesta_tarea.nota
+                         respuesta_tarea.nota,
+                         tarea.id as tarea_id
                         FROM
                           usuario
                         JOIN
                           usuario_clase ON usuario.id = usuario_clase.id_usuario
                         JOIN
-                          clase ON usuario_clase.id_clase = clase.id 
+                          clase ON usuario_clase.id_clase = clase.id
                         JOIN
                           capitulo ON clase.id = capitulo.id_clase
                         JOIN
@@ -38,9 +39,9 @@ taskRoutes.get("/chapter/:id/tasks-notes",async(req,res)=>{
                         JOIN
                           respuesta_tarea ON tarea.id = respuesta_tarea.id_tarea AND usuario.id = respuesta_tarea.id_usuario
                         WHERE
-                          clase.id =24;`;
+                          clase.id =?;`;
 
-   const [rows]=await dbConnection.query(getTasksQuery);
+   const [rows]=await dbConnection.query(getTasksQuery, [id]);
    console.log(rows)
     res.send(rows);
 })
@@ -50,7 +51,7 @@ taskRoutes.post("/chapter/:id_capitulo/tasks",async(req,res)=>{
     const {id_capitulo}=req.params;
     const {descripcion}=req.body;
 
-    const createTaskQuery=`INSERT INTO tarea (id_capitulo,descripcion) 
+    const createTaskQuery=`INSERT INTO tarea (id_capitulo,descripcion)
                             VALUES (?,?)`;
 
     await dbConnection.query(createTaskQuery,[id_capitulo,descripcion]);
@@ -61,7 +62,7 @@ taskRoutes.post("/chapter/:id_capitulo/tasks",async(req,res)=>{
 taskRoutes.delete("/chapter/:id_capitulo/tasks/:id_tarea",async(req,res)=>{
     const {id_tarea}=req.params;
 
-    const removeTaskQuery=`DELETE FROM tarea 
+    const removeTaskQuery=`DELETE FROM tarea
                            WHERE id = ?`;
 
     await dbConnection.query(removeTaskQuery,[id_tarea]);
@@ -72,8 +73,8 @@ taskRoutes.delete("/chapter/:id_capitulo/tasks/:id_tarea",async(req,res)=>{
 taskRoutes.put("/chapter/:id_capitulo/tasks/:id_tarea",async(req,res)=>{
     const {id_tarea}=req.params;
     const {descripcion}=req.body;
-    
-    const createTaskQuery=`UPDATE tarea 
+
+    const createTaskQuery=`UPDATE tarea
                            SET descripcion = ?
                            WHERE id = ?`;
     await dbConnection.query(createTaskQuery,[descripcion,id_tarea]);
@@ -88,7 +89,7 @@ taskRoutes.get("/task/:id_tarea/responses",async(req,res)=>{
                                     capitulo.*,
                                     usuario.nombre_completo
                             FROM tarea,respuesta_tarea,usuario,capitulo
-                            WHERE tarea.id=respuesta_tarea.id_tarea 
+                            WHERE tarea.id=respuesta_tarea.id_tarea
                             AND respuesta_tarea.id_usuario=usuario.id
                             AND capitulo.id=tarea.id_capitulo
                             AND tarea.id= ?`;
@@ -103,7 +104,7 @@ taskRoutes.get("/task/:id_tarea/responses",async(req,res)=>{
 taskRoutes.post("/tasks/responses",async(req,res)=>{
     const {id_usuario,id_tarea,mensaje,codigo}=req.body;
 
-    const createResponseQuery=`INSERT INTO respuesta_tarea (id_usuario,id_tarea,mensaje,codigo) 
+    const createResponseQuery=`INSERT INTO respuesta_tarea (id_usuario,id_tarea,mensaje,codigo)
                             VALUES (?,?,?,?)`;
 
     await dbConnection.query(createResponseQuery,[id_usuario,id_tarea,mensaje,codigo]);
@@ -115,8 +116,8 @@ taskRoutes.post("/tasks/responses",async(req,res)=>{
 taskRoutes.delete("/tasks/responses",async(req,res)=>{
     const {id_usuario,id_tarea}=req.body;
 
-    const deleteResponseQuery=`DELETE FROM respuesta_tarea 
-                               WHERE id_usuario = ? 
+    const deleteResponseQuery=`DELETE FROM respuesta_tarea
+                               WHERE id_usuario = ?
                                AND id_tarea = ? `;
 
     await dbConnection.query(deleteResponseQuery,[id_usuario,id_tarea]);
@@ -128,7 +129,7 @@ taskRoutes.delete("/tasks/responses",async(req,res)=>{
 taskRoutes.put("/tasks/responses",async(req,res)=>{
     const {id_usuario,id_tarea,nota}=req.body;
 
-    const updateNoteQuery=`UPDATE respuesta_tarea SET nota = ? 
+    const updateNoteQuery=`UPDATE respuesta_tarea SET nota = ?
                                WHERE id_usuario = ? AND id_tarea = ?`;
 
     await dbConnection.query(updateNoteQuery,[nota,id_usuario,id_tarea]);
@@ -146,7 +147,7 @@ taskRoutes.get("/tasks/responses/user/:id_usuario",async(req,res)=>{
                                     capitulo.*,
                                     usuario.nombre_completo
                             FROM tarea,respuesta_tarea,usuario,capitulo
-                            WHERE tarea.id=respuesta_tarea.id_tarea 
+                            WHERE tarea.id=respuesta_tarea.id_tarea
                             AND respuesta_tarea.id_usuario=usuario.id
                             AND capitulo.id=tarea.id_capitulo
                             AND usuario.id=?`;
@@ -158,4 +159,4 @@ taskRoutes.get("/tasks/responses/user/:id_usuario",async(req,res)=>{
 
 
 
-export default taskRoutes;
\ No newline at end of file
+export default taskRoutes;
diff --git a/TSS_LMS_FE/src/EducativePlatform/pages/GradesPage/GradesPage.jsx b/TSS_LMS_FE/src/EducativePlatform/pages/GradesPage/GradesPage.jsx
index 296c4b1..b776db3 100644
--- a/TSS_LMS_FE/src/EducativePlatform/pages/GradesPage/GradesPage.jsx
+++ b/TSS_LMS_FE/src/EducativePlatform/pages/GradesPage/GradesPage.jsx
@@ -15,6 +15,12 @@ export const GradesPage = () => {
 
     const calculateAverages = (data) => {
         const userGrades = {};
+        const differentTasks = [];
+        data.forEach((task) => {
+            if (!differentTasks.includes(task.tarea_id)) {
+                differentTasks.push(task.tarea_id);
+            }
+        });
 
         data.forEach(student => {
             const { id, nota } = student;
@@ -24,14 +30,13 @@ export const GradesPage = () => {
             }
 
             userGrades[id].sum += nota;
-            userGrades[id].count += 1;
         });
 
         const averages = Object.keys(userGrades).map(userId => ({
             id: userId,
             nombre_completo: userGrades[userId].nombre_completo,
             email: userGrades[userId].email,
-            promedio: userGrades[userId].sum / userGrades[userId].count
+            promedio: userGrades[userId].sum / differentTasks.length,
         }));
 
         setAverages(averages);
